packages/itwinui-react/src/core/FileUpload/FileUploadCard.test.tsx

```diff
@@ -0,0 +1,144 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { SvgDocument, SvgUpload, SvgCheckmark } from '../utils';
+import { FileUploadCard, FileUploadCardProps } from './FileUploadCard';
+import { Button } from '../Buttons';
+
+const CustomFileUploadCard = (props: FileUploadCardProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const [files, setFiles] = React.useState<File[]>(
+    new Array<File>(
+      new File(['mock-file'], 'test.txt', { type: 'text/plain' }),
+    ),
+  );
+
+  return (
+    <FileUploadCard
+      files={files}
+      onFilesChange={(files) => setFiles(files)}
+      {...props}
+    >
+      <FileUploadCard.Icon>
+        <SvgCheckmark />
+      </FileUploadCard.Icon>
+      <FileUploadCard.Text>
+        <FileUploadCard.Label>TestLabel</FileUploadCard.Label>
+        <FileUploadCard.Description>TestDescription</FileUploadCard.Description>
+      </FileUploadCard.Text>
+      <FileUploadCard.Action>
+        <Button
+          onClick={() => {
+            setFiles([]);
+          }}
+        >
+          Clear
+        </Button>
+        <FileUploadCard.Input ref={inputRef} />
+      </FileUploadCard.Action>
+    </FileUploadCard>
+  );
+};
+
+it('should render empty FileUploadCard before a file is uploaded', () => {
+  const { container } = render(<FileUploadCard />);
```

Comment:
All of the unit tests are failing because of this error in `FileUploadCardInput` that I'm not quite sure how to fix yet

![image](https://user-images.githubusercontent.com/78564221/216196571-2fe98604-555e-4f9e-9407-8392f4a6e4e5.png)

Reply:
The issue is that DataTransfer is not in jest, so it must be mocked. I've changed things so that the tests pass, but I still need to properly mock DataTransfer in these unit tests to make sure it's working

Reply:
I worked on this with Mayank, and we came to the conclusion that it's not worth mocking `DataTransfer` in jest just to get tests to pass, so I have written a check for `DataTransfer` in `FileUploadCard` so the tests can still pass